src/lambda_function/s3_to_glue/app.py

```diff
@@ -12,14 +12,56 @@
 logger = logging.getLogger()
 logger.setLevel(logging.INFO)
 
-def submit_s3_to_json_workflow(messages: list[dict[str,str]], workflow_name):
+
+def filter_object_info(object_info: dict) -> dict:
+    """Filter out object info records that should not be processed. Returns None
+
+       - Records containing owner.txt
+       - Records that don't contain a specific object key like <path_to_file>/<file_name>
+       - Records that are missing a source_key
+       - Records that are missing a bucket_name
+
+    Args:
+        object_info (dict): Object information from source S3 bucket
+
+    Returns:
+        dict: returns back the object info dict if it passes the filter criteria
+            otherwise returns None
+    """
+    if not object_info["source_key"]:
+        logger.info(
+            "This object_info record doesn't contain a source key "
+            f"and can't be processed.\nMessage: {object_info}",
+        )
+        return None
+    elif not object_info["source_bucket"]:
+        logger.info(
+            "This object_info record doesn't contain a source bucket "
+            f"and can't be processed.\nMessage: {object_info}",
+        )
+        return None
+    elif "owner.txt" in object_info["source_key"]:
+        logger.info(
+            f"This object_info record is an owner.txt and can't be processed.\nMessage: {object_info}"
+        )
+        return None
+    elif object_info["source_key"].endswith("/"):
+        logger.info(
+            f"This object_info record is a directory and can't be processed.\nMessage: {object_info}"
+        )
+        return None
```

Comment:
Nit: You may be able to simplify all these returns by just having an end return, BUT i like that you're being explicit here.

```
else:
    return object_info
return None
```